Documentation/installing-dependencies.rst

```diff
@@ -1,216 +1,293 @@
+  ln -s ${PREFIX}/bin/cmake/bin/cmake ${PREFIX}/bin
 
-Note that this extracts CMake to the directory ${HOME}/bin/cmake, if you wish you can adjust that path.
+Note that this extracts CMake to the directory ``${PREFIX}/bin/cmake``, if you wish you can adjust that path. Note that you may now also use ``ccmake`` to get a terminal UI for configuring the following libraries.
   
+Required Libraries
+~~~~~~~~~~~~~~~~~~
+
+The following libraries need to installed for all SeisSol CPU and GPU builds.
+To get a working CPU build, installing all libraries described here is enough.
+However, installing a GEMM generator and a graph partitioner is still recommended for better performance and better load balancing, respectively.
+
 Installing HDF5
-~~~~~~~~~~~~~~~
+"""""""""""""""
+
+We begin with HDF5 which is needed for reading meshes in PUML format (the default format in SeisSol, and the output of PUMgen).
+If your system does not have it e.g. as a module file (type `module avail | grep hdf5` to look for it),
+you may compile it manually with the following commands:
 
 .. code-block:: bash
 
   wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.8/src/hdf5-1.10.8.tar.bz2
   tar -xaf hdf5-1.10.8.tar.bz2
   cd hdf5-1.10.8
-  CPPFLAGS="-fPIC ${CPPFLAGS}" CC=mpicc FC=mpif90 ./configure --enable-parallel --prefix=$HOME --with-zlib --disable-shared --enable-fortran 
+  CPPFLAGS="-fPIC ${CPPFLAGS}" CC=mpicc CXX=mpicxx ./configure --enable-parallel --prefix=$PREFIX --with-zlib --disable-shared
   make -j8
   make install
   cd ..
 
-Installing netCDF
-~~~~~~~~~~~~~~~~~
+Installing Eigen
+""""""""""""""""
 
-.. code-block:: bash
-
-  wget https://syncandshare.lrz.de/dl/fiJNAokgbe2vNU66Ru17DAjT/netcdf-4.6.1.tar.gz
-  tar -xaf netcdf-4.6.1.tar.gz
-  cd netcdf-4.6.1
-  CFLAGS="-fPIC ${CFLAGS}" CC=h5pcc ./configure --enable-shared=no --prefix=$HOME --disable-dap
-  #NOTE: Check for this line to make sure netCDF is built with parallel I/O: 
-  #"checking whether parallel I/O features are to be included... yes" This line comes at the very end (last 50 lines of configure run)!
-  make -j8
-  make install
-  cd ..
-
-.. _installing_eigen3:
-
-Installing Eigen3
-~~~~~~~~~~~~~~~~~
+Next, we look at Eigen which conveniently uses CMake as a build system for itself.
+Eigen is used in SeisSol for setting up matrices and other numerical computations, and optionally, also as code generator for matrix chain products.
+Once again, if you do not have Eigen installed, you may do so manually as follows:
 
 .. code-block:: bash
 
    wget https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
    tar -xf eigen-3.4.0.tar.gz
    cd eigen-3.4.0
    mkdir build && cd build
-   cmake .. -DCMAKE_INSTALL_PREFIX=~
+   cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX
    make install
    cd ../..
 
-.. _installing_libxsmm:
+Lastly, we need easi which is (most likely) not already installed on your system or as a module file, as it is a more SeisSol-specific library.
```

Comment:
Easi is used for setting up the model parameters.
It is (most likely) not already installed on your system or as a module file, as it is a more SeisSol-specific library.